### ISSUE TITLE
[HOLD for payment 2023-07-26] [$1000] The chat composer is hidden and is not scrollable on a new page with a description/date

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#19389]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Go to staging dot on mweb chrome
2. Click on + icon
3. Select send money > enter amount
4. Select a recipient > send the payment
5. Click `I'll settle up elsewhere`
6. Click on the payment request in the DM of the recipient
7. This takes you to a new screen where you can write a message under the payment request
8. Click on chat composer so that keyboard opens up
9. Notice that the top part of the composer is hidden
10. Additionally, if you type a long description, you can't see it 
11. Also, you're unable to scroll the long description while the keyboard is open

## Expected Result:
The chat composer should not be hidden on a new page with description and date after sending money
## Actual Result:
The chat composer is hidden on a new page with description and date after sending money

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.16-6
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/a6cd5295-d937-43d0-b5d7-5be844d46f57

https://github.com/Expensify/App/assets/43996225/e2f486f2-6829-49d5-8550-69607e7810fb

**Expensify/Expensify Issue URL:**
**Issue reported by:** @priya-zha
**Slack conversation:**  https://expensify.slack.com/archives/C049HHMV9SM/p1684475099942009

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01beba9fa9a0c7d961</li>
        <li>Upwork Job ID: 1661041516743192576</li>
        <li>Last Price Increase: 2023-06-13</li>
    </ul>
</details>

**Comment by NicMendonca** [comment]
Going OOO until June 5th so assigning a buddy to this GH to watch over the issue. I'll pick it back up when I am back from OOO/ if this is still open.

**Comment by akinwale** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The chat composer is hidden and not scrollable on the money request page.

### What is the root cause of that problem?
The MoneyRequestHeader component occupies too much space on mobile screens, which causes it to overlap and hide the chat area when the soft keyboard is visible.

### What changes do you think we should make in order to solve the problem?
Auto-layout the MoneyRequestHeader to rearrange and display the information for more compact screens. The header could be conditionally resized based on whether or not the soft keyboard is visible as shown in the video demo.

### What alternative solutions did you explore? (Optional)
Alternatively, always display a compact version of the money request header for mobile screens.


https://github.com/Expensify/App/assets/4160319/c6f93e1c-c910-46bb-b39d-e62da05a60cd

**Comment by maddylewis** [comment]
i'll let @aimane-chnaif review this most recent proposal and confirm if that's how we should move forward.

**Comment by madmax330** [comment]
Waiting on proposal reviews @aimane-chnaif

**Comment by maddylewis** [comment]
@aimane-chnaif - can you link the other slack thread that's referenced [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1685604943195569?thread_ts=1685553199.009849&cid=C01GTK53T8Q) to this issue? thanks!

**Comment by madmax330** [comment]
@aimane-chnaif looks like there's a consensus on how to move forward in the thread you linked. Seems like the solution is to make it scrollable.

**Comment by NicMendonca** [comment]
@maddylewis I'm back 👋 unassigning you

**Comment by ShogunFire** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

The screen to see a money request is not scrollable and the keyboard is on top of the date

### What is the root cause of that problem?

We haven't made the screen scrollable

### What changes do you think we should make in order to solve the problem?


To make the screen scrollable we need to add a ScrollView element to MoneyRequestHeader. We can put it surrounding the existing root element [here](https://github.com/Expensify/App/blob/6e7f460b9d8853c25e665668452a9a8098eb6176/src/components/MoneyRequestHeader.js#L94)

This is the result:
![image](https://github.com/Expensify/App/assets/19537677/c47840d8-5173-4a4f-ac4f-ad05ddcf0819)


### What alternative solutions did you explore? (Optional)

We can put the scrollview surrounding different elements depending what we want to be able to see at all time

**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by ShogunFire** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

The header overlaps the input when the keyboard is open in the money request report

### What is the root cause of that problem?

The size of the header + keyboard + input is bigger than the size of the screen and the header is fixed at the top instead of being scrollable

### What changes do you think we should make in order to solve the problem?


Right now MoneyRequestHeader is in ReportScreen [here](https://github.com/Expensify/App/blob/4cad2489f6feb5efb955924764711a0c36439f58/src/pages/home/ReportScreen.js#L266) , to make the MoneyRequestHeader not fixed at the top we can move it to ReportActionsView [here](https://github.com/Expensify/App/blob/4cad2489f6feb5efb955924764711a0c36439f58/src/pages/home/report/ReportActionsView.js#L341)

This is the result:

https://github.com/Expensify/App/assets/19537677/8bfa21cb-c56f-4850-a256-b4086e4f6269




### What alternative solutions did you explore? (Optional)


**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by NicMendonca** [comment]
@aimane-chnaif any feedback on [this proposal](https://github.com/Expensify/App/issues/19389#issuecomment-1582516671)?

**Comment by ShogunFire** [comment]
IOS:

https://github.com/Expensify/App/assets/19537677/7f55dce6-179b-48f8-9daa-c4251cf4f02c

Android:

https://github.com/Expensify/App/assets/19537677/7816002b-21ad-48c3-bc00-efd43a4c09da

Safari:

https://github.com/Expensify/App/assets/19537677/28e25e53-d74c-4946-8258-9d1a880c025e


The safari one appears weirdly but I thinkit's only because of the not secure thing, as you can see it also appear weird on the other reports

**Comment by akinwale** [comment]
For anyone who wishes to tackle this, the best approach would've been to separate the `MenuItemWithTopDescription` components displayed for the single transaction view into a new component, and set that component as the `ListHeaderComponent` on the `InvertedFlatList` in the `ReportActionsList` component. However, with the underlying `FlatList` having `inverted` set to `true`, items are being rendered from bottom to top. Unfortunately, this also affects the "header" component (see screenshot below), and I think it would require a significant rework / overhaul of the way report actions data is being arranged / displayed to get it to work.

![Screenshot 2023-06-09 at 08 37 00](https://github.com/Expensify/App/assets/4160319/968f1518-b482-4a81-9869-a6c357f3ca59)

**Comment by ShogunFire** [comment]
> @ShogunFire I don't see any difference between current version and your proposal. Money request header should be scrollable. Only top-most header should be fixed.

In this version the top most header is fixed... I thought you were just talking about the first line above To.
The rest of the header is not scrollable by hand gesture but it goes up when the keyboard opens which makes the input seeable. Is it not enough ?

**Comment by ShogunFire** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/19389#issuecomment-1582516671)

**Comment by ShogunFire** [comment]
When the list is empty:

https://github.com/Expensify/App/assets/19537677/7998544b-6fb1-42b2-bd7c-ced58bd86988

**Comment by ShogunFire** [comment]
https://github.com/Expensify/App/assets/19537677/dc3fe5b4-aa1e-4c2f-9ff2-4e3cf4111506

**Comment by ShogunFire** [comment]
Hello @aimane-chnaif  I have updated the video, I think this time it is what we want. The PR is ready if you want to assign me

**Comment by madmax330** [comment]
@aimane-chnaif can you look at the updated proposals?

**Comment by NicMendonca** [comment]
@aimane-chnaif bump!

**Comment by ShogunFire** [comment]
@aimane-chnaif https://github.com/ShogunFire/App/tree/moneyRequestHeaderScrollable

**Comment by madmax330** [comment]
@aimane-chnaif can you review if any of the proposals were ok?

**Comment by ShogunFire** [comment]
@aimane-chnaif  I merged main into my branch and tested on all platforms

### Screenshots/Videos
<details>
<summary>Web</summary>



https://github.com/Expensify/App/assets/19537677/c39900c9-3e8a-4571-895d-86f9ca6686af


</details>

<details>
<summary>Mobile Web - Chrome</summary>



https://github.com/Expensify/App/assets/19537677/b5a2e6d4-257f-4cac-a021-c3af3b2011ef


</details>

<details>
<summary>Mobile Web - Safari</summary>



https://github.com/Expensify/App/assets/19537677/49cfeb82-1828-4594-af52-c04b1f090b12


</details>

<details>
<summary>Desktop</summary>



https://github.com/Expensify/App/assets/19537677/88f3e01f-c3ac-4909-9287-c0eeadc46141


</details>

<details>
<summary>iOS</summary>



https://github.com/Expensify/App/assets/19537677/3b1800d8-53d1-4de8-a577-75c7b5fe7dde


</details>

<details>
<summary>Android</summary>



https://github.com/Expensify/App/assets/19537677/d51fac2e-a08e-4dd5-9824-2377471c3ae6


</details>

**Comment by MaciejSWM** [comment]
Hey guys! I'll be taking a shot on this. I'm an external from SWM.

**Comment by madmax330** [comment]
@aimane-chnaif @MaciejSWM  can you look at this latest proposal? https://github.com/Expensify/App/issues/19389#issuecomment-1599207981

If it is not satisfactory I will assign to @MaciejSWM

**Comment by NicMendonca** [comment]
@kevinksullivan I am going OOO until Wednesday. Can you watch this while I am away? I'll unassign you when I am back. Thank you!

**Comment by ShogunFire** [comment]
Hello, looks like it has been made internal for some reason, I don't have access to https://stackoverflow.com/c/expensify/questions/14418  so can someone just tell me why since I worked quite a bit on this and was very close to a solution I think

**Comment by kevinksullivan** [comment]
Hi @aimane-chnaif @MaciejSWM let us know what you think so we can decide the direction here. Thanks!

**Comment by MaciejSWM** [comment]
@kevinksullivan I'm on it right now. @ShogunFire's solution fixes the issue for me. I'm jumping into the code now to see how it's implemented.

**Comment by MaciejSWM** [comment]
I also tested on all platforms and had no problems there.

**Comment by kevinksullivan** [comment]
Thanks @MaciejSWM !

**Comment by NicMendonca** [comment]
@kevinksullivan I'm back!

**Comment by ShogunFire** [comment]
Thanks for the review @MaciejSWM 
Can you assign me @aimane-chnaif if it's also good for you ?

**Comment by NicMendonca** [comment]
@MaciejSWM @ShogunFire assigned you both 🚀

**Comment by ShogunFire** [comment]
Ready for review @aimane-chnaif

**Comment by madmax330** [comment]
Yeah LGTM

**Comment by ShogunFire** [comment]
Just to not forget, I think the paiement date should be the 26th since there was one regression that will achieve 7 days in production the 26

**Comment by NicMendonca** [comment]
^ sorry, was just checking to see if that would re-create the job in Upwork since the original job post expired 👼

**Comment by NicMendonca** [comment]
@ShogunFire here's the job post! https://www.upwork.com/jobs/~016e8237e6e708015c

@aimane-chnaif are you requesting payment via Expensify?

**Comment by ShogunFire** [comment]
@NicMendonca Sorry but can you invite me ? I have already applied on another issue so I think they have stored my profile for next time and I don't want to use 16 connects on all my current issues

**Comment by NicMendonca** [comment]
@ShogunFire I can't find you to invite, you are not searchable via GH username. Can you please link your Upwork or apply to the job here: https://www.upwork.com/jobs/~016e8237e6e708015c

@aimane-chnaif thanks! Just sent you the offer

**Comment by ShogunFire** [comment]
https://www.upwork.com/freelancers/pierremichel5

**Comment by NicMendonca** [comment]
@aimane-chnaif you've been paid 🎉 

@ShogunFire just need you to accept the offer, please

**Comment by NicMendonca** [comment]
@aimane-chnaif bump on the BZ checklist also: https://github.com/Expensify/App/issues/19389#issuecomment-1633406312

**Comment by NicMendonca** [comment]
@ShogunFire you've been paid! 🎉

**Comment by NicMendonca** [comment]
everyone has been paid!

**Comment by priya-zha** [comment]
@NicMendonca I'm the external reporter. I think I haven't been paid for this yet.

**Comment by NicMendonca** [comment]
@priya-zha sorry! I was getting confused by another GH. 

Can you please apply to the job? https://www.upwork.com/jobs/~016e8237e6e708015c

**Comment by priya-zha** [comment]
@NicMendonca submitted the proposal. Thanks

**Comment by ShogunFire** [comment]
@NicMendonca Hello, sorry I just noticed that I have accepted the offer but the payment hasn't been made. 

Here is my profile again: https://www.upwork.com/freelancers/pierremichel5

**Comment by ShogunFire** [comment]
@NicMendonca Hello, sorry for the late request but could you put the upwork mission as completed ? Because it makes my upwork profile worse otherwise. Thanks

**Comment by stephanieelliott** [comment]
> Hello, sorry for the late request but could you put the upwork mission as completed ? Because it makes my upwork profile worse otherwise. Thanks

I took care of this!
